codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java

```diff
@@ -0,0 +1,825 @@
+    }
+
+    @Override
+    public boolean contains(CharSequence name) {
+        return get(name) != null;
+    }
+
+    @Override
+    public boolean contains(CharSequence name, CharSequence value) {
+        final int nameHash = AsciiString.hashCode(name);
+        final int valueHash = AsciiString.hashCode(value);
+
+        final int pseudoHeadersEnd = pseudoHeaders.length - 1;
+        for (int i = 0; i < pseudoHeadersEnd; i += 2) {
+            AsciiString roName = pseudoHeaders[i];
+            AsciiString roValue = pseudoHeaders[i + 1];
+            if (roName.hashCode() == nameHash && roValue.hashCode() == valueHash &&
+                roName.contentEqualsIgnoreCase(name) && roValue.contentEqualsIgnoreCase(value)) {
+                return true;
+            }
+        }
+
+        final int otherHeadersEnd = otherHeaders.length - 1;
+        for (int i = 0; i < otherHeadersEnd; i += 2) {
+            AsciiString roName = otherHeaders[i];
+            AsciiString roValue = otherHeaders[i + 1];
+            if (roName.hashCode() == nameHash && roValue.hashCode() == valueHash &&
+                roName.contentEqualsIgnoreCase(name) && roValue.contentEqualsIgnoreCase(value)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean containsObject(CharSequence name, Object value) {
+        if (value instanceof CharSequence) {
+            return contains(name, (CharSequence) value);
+        }
+        return contains(name, value.toString());
+    }
+
+    @Override
+    public boolean containsBoolean(CharSequence name, boolean value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsByte(CharSequence name, byte value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsChar(CharSequence name, char value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsShort(CharSequence name, short value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsInt(CharSequence name, int value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsLong(CharSequence name, long value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsFloat(CharSequence name, float value) {
+        return false;
+    }
+
+    @Override
+    public boolean containsDouble(CharSequence name, double value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public boolean containsTimeMillis(CharSequence name, long value) {
+        return contains(name, String.valueOf(value));
+    }
+
+    @Override
+    public int size() {
+        return (pseudoHeaders.length + otherHeaders.length) >>> 1;
```

Comment:
actually, why not do it the same way binary search implementations avoid overflows:
```
return pseudoHeaders.length /2 + otherHeaders.length / 2;
```